internal/controller/object/object.go

```diff
@@ -0,0 +1,310 @@
+	}
+
+	return &external{
+		logger: c.logger,
+		client: resource.ClientApplicator{
+			Client:     k,
+			Applicator: resource.NewAPIPatchingApplicator(k),
+		},
+	}, nil
+}
+
+type external struct {
+	logger logging.Logger
+	client resource.ClientApplicator
+}
+
+func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Object)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotKubernetesObject)
+	}
+
+	c.logger.Debug("Observing", "resource", cr)
+
+	desired, err := getDesired(cr)
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	observed := desired.DeepCopy()
+
+	err = c.client.Get(ctx, types.NamespacedName{
+		Namespace: observed.GetNamespace(),
+		Name:      observed.GetName(),
+	}, observed)
+
+	if kerrors.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetObject)
+	}
+
+	if err = setObserved(cr, observed); err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	var last *unstructured.Unstructured
+	if last, err = getLastApplied(cr, observed); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetLastApplied)
+	}
+	if last == nil {
+		return managed.ExternalObservation{
+			ResourceExists:   true,
+			ResourceUpToDate: false,
+		}, nil
+	}
+
+	if equality.Semantic.DeepEqual(last, desired) {
+		c.logger.Debug("Up to date!")
+		return managed.ExternalObservation{
+			ResourceExists:   true,
+			ResourceUpToDate: true,
+		}, nil
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: false,
+	}, nil
+}
+
+func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Object)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotKubernetesObject)
+	}
+
+	c.logger.Debug("Creating", "resource", cr)
+	obj, err := getDesired(cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	obj.SetAnnotations(map[string]string{
+		v1.LastAppliedConfigAnnotation: string(cr.Spec.ForProvider.Manifest.Raw),
+	})
+	if err := c.client.Create(ctx, obj); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateObject)
+	}
+
+	cr.Status.SetConditions(xpv1.Available())
+	return managed.ExternalCreation{}, setObserved(cr, obj)
+}
+
+func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.Object)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotKubernetesObject)
+	}
+
+	c.logger.Debug("Updating", "resource", cr)
+
+	obj, err := getDesired(cr)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	obj.SetAnnotations(map[string]string{
+		v1.LastAppliedConfigAnnotation: string(cr.Spec.ForProvider.Manifest.Raw),
+	})
```

Comment:
@turkenh does this cause any conflict with k8s trying to set this / will k8s not do this automatically?

Reply:
k8s does not set this automatically rather clients like `kubectl apply` set this. I also noticed ArgoCD also sets the same.
So, no conflict as long as someone kubectl apply's on the object, which would be reverted back by the provider-kubernetes controller.